pymc3/plots/scatterplot.py

```diff
@@ -0,0 +1,136 @@
+
+    """
+    Plot a scatter or hexbin matrix of the sampled parameters.
+
+    Parameters
+    ----------
+
+    trace : result of MCMC run
+    varnames : list of variable names
+        Variables to be plotted, if None all variable are plotted
+    figsize : figure size tuple
+        If None, size is (8 + numvars, 8 + numvars)
+    text_size: int
+        Text size for labels
+    ax : axes
+        Matplotlib axes.
+    hexbin : Boolean
+        If True draws an hexbin plot
+    plot_transformed : bool
+        Flag for plotting automatically transformed variables in addition to
+        original variables (defaults to False). Applies when varnames/sub_varnames = None.
+        When a list of varnames/sub_varnames is passed, transformed variables can be passed
+        using their names.
+    divergences : Boolean
+        If True divergences will be plotted in a diferent color
+    kwargs_divergence : dicts, optional
+        Aditional keywords passed to ax.scatter for divergences
+    sub_varnames : list, optional        
+        Aditional varnames passed for plotting subsets of multidimensional variables
+    Returns
+    -------
+
+    ax : matplotlib axes
+
+    """    
+    
+    if varnames is None:
+        if plot_transformed:
+            trace_dict = get_trace_dict(trace, get_default_varnames(trace.varnames, True))
+        else:
+            trace_dict = get_trace_dict(trace, get_default_varnames(trace.varnames, False))
+        if sub_varnames is None:
+            varnames = list(trace_dict.keys())
+        else:
+            varnames = sub_varnames
+    else:
+        trace_dict = get_trace_dict(trace, varnames)
+        varnames = list(trace_dict.keys())       
+
+    if text_size is None:
+        text_size = scale_text(figsize, text_size=text_size)
+
+    if kwargs_divergence is None:
+        kwargs_divergence = {}
+    
+    numvars = len(varnames)
+
+    if figsize is None:
+        figsize = (8+numvars, 8+numvars)    
+
+    if numvars < 2:
+        raise Exception('Number of variables to be plotted must be 2 or greater.')
+        
+    if numvars == 2 and ax is not None:
+        if hexbin:
+            ax.hexbin(trace_dict[varnames[0]], trace_dict[varnames[1]], mincnt=1, **kwargs)
+        else:
+            ax.scatter(trace_dict[varnames[0]], trace_dict[varnames[1]], **kwargs)
+        
+        if divergences:
+            try:
+                divergent = trace['diverging']
+                if np.any(divergent):
+                    ax.scatter(trace_dict[varnames[0]][divergent == 1], trace_dict[varnames[1]][divergent == 1], **kwargs_divergence)
+                else:
+                    print('No divergences were found.')
+            except KeyError:
+                warnings.warn('There is no divergence information in the passed trace.')
+                return ax
+
+    if ax is None:
+        _, ax = plt.subplots(nrows=numvars, ncols=numvars, figsize=figsize)
```

Comment:
if you do `nrows=numvars - 1, ncols=numvars - 1`, then the two loops below would be

```
for i in range(numvars):
    for j in range(i + 1, numvars):
        ...
```
and you can remove the `if i == j` block

(this might be totally off base, if `axes.remove()` does not do what I think it does).